crates/flux-refineck/src/constraint_gen.rs

```diff
@@ -583,7 +583,8 @@ impl<'a, 'genv, 'tcx> InferCtxt<'a, 'genv, 'tcx> {
             (BaseTy::Bool, BaseTy::Bool)
             | (BaseTy::Str, BaseTy::Str)
             | (BaseTy::Char, BaseTy::Char)
-            | (BaseTy::RawPtr(_, _), BaseTy::RawPtr(_, _)) => Ok(()),
+            | (BaseTy::RawPtr(_, _), BaseTy::RawPtr(_, _))
+            | (BaseTy::Dynamic(_, _), BaseTy::Dynamic(_, _)) => Ok(()),
```

Comment:
I'd add an assertion here checking both types are the same. I'm not sure if there's something odd in Rust regarding subtyping between dynamic objects so it's better to be safe (I suppose the lifetime can be different, but the existential predicate have to be the same)

Reply:
done!